test/integration/controller_binding_test.go

```diff
@@ -0,0 +1,554 @@
+	_ "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/install"
+
+	osb "github.com/pmorie/go-open-service-broker-client/v2"
+	fakeosb "github.com/pmorie/go-open-service-broker-client/v2/fake"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/test/util"
+	"net/http"
+)
+
+// TestCreateServiceBindingSuccess successful paths binding
+func TestCreateServiceBindingSuccess(t *testing.T) {
+	cases := []struct {
+		name string
+	}{
+		{
+			name: "defaults",
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ct := &controllerTest{
+				t:        t,
+				broker:   getTestBroker(),
+				instance: getTestInstance(),
+				binding:  getTestBinding(),
+			}
+			ct.run(func(ct *controllerTest) {
+				{
+					condition := v1beta1.ServiceBindingCondition{
+						Type:   v1beta1.ServiceBindingConditionReady,
+						Status: v1beta1.ConditionTrue,
+					}
+					if cond, err := util.WaitForBindingConditionLastSeenOfType(ct.client, testNamespace, testBindingName, condition); err != nil {
+						t.Fatalf("error waiting for binding condition: %v\n"+"expecting: %+v\n"+"last seen: %+v", err, condition, cond)
+					}
+				}
+			})
+		})
+	}
+}
+
+// TestCreateServiceBindingInvalidInstance try to bind to invalid service instance names
+func TestCreateServiceBindingInvalidInstance(t *testing.T) {
+	cases := []struct {
+		name                string
+		instanceName        *string
+		expectedErrorReason string
+		expectedFailure     bool
+	}{
+		{
+			name:                "non-existent service instance name",
+			instanceName:        strPtr("nothereinstance"),
+			expectedErrorReason: "ReferencesNonexistentInstance",
+		},
+		{
+			name:                "invalid service instance name",
+			instanceName:        strPtr(""),
+			expectedErrorReason: "ReferencesNonexistentInstance",
+			expectedFailure:     true,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ct := &controllerTest{
+				t: t,
+				skipBindingCreateError: tc.expectedFailure,
+				broker:                 getTestBroker(),
+				instance:               getTestInstance(),
+				binding: func() *v1beta1.ServiceBinding {
+					b := getTestBinding()
+					if tc.instanceName != nil {
+						b.Spec.ServiceInstanceRef.Name = *tc.instanceName
+					}
+					return b
+				}(),
+				skipVerifyingBindingSuccess: tc.expectedErrorReason != "",
+			}
+			ct.run(func(ct *controllerTest) {
+				{
+					status := v1beta1.ConditionTrue
+					if tc.expectedErrorReason != "" {
+						status = v1beta1.ConditionFalse
+					}
+					condition := v1beta1.ServiceBindingCondition{
+						Type:   v1beta1.ServiceBindingConditionReady,
+						Status: status,
+						Reason: tc.expectedErrorReason,
+					}
+					if ct.skipBindingCreateError {
+						if err := util.WaitForBindingToNotExist(ct.client, testNamespace, testBindingName); err != nil {
```

Comment:
I don't think that this is exactly what we want to test. It is a bit convoluted to wait for something that never existed to now not exist. What we really want to test is that the call to `Create` returned an error. But we've hidden that inside controllerTest. I suggest that in the cases where you expect `Create` to fail, that you call create directly rather than having controllerTest call create on your behalf. See https://github.com/staebler/service-catalog/blob/cdb2ac7a6ca10f2958dd16c2b42372d7937ce957/test/integration/controller_instance_test.go#L802.